analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/RuleChecks/EmptyCollectionsShouldNotBeEnumeratedBase.cs

```diff
@@ -130,6 +130,10 @@ protected override ProgramState PreProcessSimple(SymbolicContext context)
         {
             return ProcessMethod(context, invocation.TargetMethod, invocation.Instance);
         }
+        else if (operation.AsPropertyReference() is { Property.IsIndexer: true } propertyReference)
+        {
+            return ProcessIndexerAccess(context.State, propertyReference);
+        }
```

Comment:
Invocation above and method reference blow share (almost) the same logic. So this should be below, not to break the flow